### PR TITLE
[CALCITE-6205] Add BITAND_AGG, BITOR_AGG functions (enabled in Snowfl…

### PR DESCRIPTION
**Pull request by tanclary**
…ake library)

### CODE REVIEW

**Review by mihaibudiu — APPROVED** [review]
I find it a bit ironic that we have aggregation functions with bitwise operations but we have no corresponding scalar functions that do the same thing.

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 1]
```diff
@@ -6662,6 +6662,28 @@ private void checkLiteral2(String expression, String expected) {
...
+    sql(query).withLibrary(SqlLibrary.SNOWFLAKE).withSnowflake().ok(expectedSnowflake);
+  }
+
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6205">[CALCITE-6205]
+   * Add BITAND_AGG, BITOR_AGG functions (enabled in Snowflake library)</a>. */
+  @Test void testBitOrAgg() {
+    final String query = "select bit_or(\"product_id\")\n"
+        + "from \"product\"";
+    final String expectedSnowflake = "SELECT BITOR_AGG(\"product_id\")\n"
```
Comment (JiajunBernoulli) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:6682]: Should we replace **BITOR_AGG** to **BITOR** when `unparse` BigQuery sql?

File: core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java [hunk 2]
```diff
@@ -1793,7 +1797,7 @@ static class BitOpImplementor extends StrictAggImplementor {
       if (SqlTypeUtil.isBinary(info.returnRelType())) {
         start = Expressions.field(null, ByteString.class, "EMPTY");
       } else {
-        Object initValue = info.aggregation() == BIT_AND ? -1L : 0;
+        Object initValue = info.aggregation().kind == SqlKind.BIT_AND ? -1L : 0;
```
Comment (olivrlee) [inline core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java:1800]: This change is just to be consistent with the other functions right?

Comment (tanclary) [reply core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java:1800]: This change is to make sure "BITAND_AGG" and "BIT_AND" are caught, previously it was just checking for a specific operator so "BITAND_AGG" was not going down the correct codepath

File: core/src/main/java/org/apache/calcite/sql/fun/SqlBitOpAggFunction.java [hunk 3]
```diff
@@ -39,7 +39,7 @@ public class SqlBitOpAggFunction extends SqlAggFunction {
 
   //~ Constructors -----------------------------------------------------------
 
-  /** Creates a SqlBitOpAggFunction. */
+  /** Creates a SqlBitOpAggFunction from a SqlKind. */
```
Comment (olivrlee) [inline core/src/main/java/org/apache/calcite/sql/fun/SqlBitOpAggFunction.java:42]: nit: What do you think about changing the previous call sites of this to call the new function with 
`SqlBitOpAggFunction(call.getKind().getName(), call.getKind())`?

Comment (tanclary) [reply core/src/main/java/org/apache/calcite/sql/fun/SqlBitOpAggFunction.java:42]: I'm indifferent, the standard operators can derive the name from the `SqlKind`, the library-specific ones cannot so they need the name `specified`. Whether it's refactored so they can use the same constructor doesn't matter too much to me

Comment (tanclary) [reply core/src/main/java/org/apache/calcite/sql/fun/SqlBitOpAggFunction.java:42]: Actually re-reading this I think I would prefer keeping the `SqlBitOpAggFunction(SqlKind kind)` constructor, I think it makes sense to only pass in one parameter if possible.

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java (hunk 1 above)
Comment (tanclary) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:6682]: My JIRA case was actually wrong, `BIT_OR` and `BIT_AND` are both standard operators not specific to BigQuery, I updated the case.

**Comment by tanclary** [comment]
Are there any other comments @JiajunBernoulli @olivrlee @mihaibudiu ? If not, I will probably merge this within the next day or so.

**Review by mihaibudiu — APPROVED** [review]

**Review by JiajunBernoulli — APPROVED** [review]

**Pull request merged**